utils/weights.py

```diff
@@ -1,17 +1,174 @@
-def get_weights():
+
+from utils.const import DEFAULT_CONTRACTED_HOURS
+from gurobipy.gurobipy import tupledict
+
+# Set weighting of Optimality Model
+
+def set_weights():
+    """
+    Fairness aspects are translated into how many hours of demand deviation that is accepted to satisfy the respective
+    fairness aspect. In addition, both the weight of the least favoured employee fairness score and the scaling between
+    excess and deficit demand is set. The latter makes it possible to increase or decrease the impact of either
+    over- or under staffing.
+    """
+
+    return{
+        # Weight of fairness aspects. Rest, contracted hours and preferences should be treated as hours.
+        "rest": 20,
+        "contracted hours": 2,
+        "partial weekends": 8,
+        "isolated working days": 10,
+        "isolated off days": 10,
+        "consecutive days": 12,
+        "preferences": 1.5,
+
+        # Weight of least favored employee. Use interval [0, 1].
+        "lowest fairness score": 0.5,
+
+        # Scaling excess and deficit of demand
+        "excess demand deviation factor": 1.0,
+        "deficit demand deviation factor": 1.0,
+    }
+
+
+# Weighting of Shift Design Model
+
+def get_shift_design_weights():
     return {
-        "rest": 5,
-        "contracted hours": 1,
-        "partial weekends": 1,
-        "isolated working days": 1,
-        "isolated off days": 1,
-        "consecutive days": 1,
-        "backward rotation": 1,
-        "preferences": 1,
-        "lowest fairness score": 1,
         "demand_deviation": 5,
+        "shift_duration": 15,
```

Comment:
Is this actually a weight?

Reply:
Bad naming. Should represent the weighting of shifts that are either longer or shorter than the desired shift duration. Do you have a possible better name for it?

Reply:
What about splitting it into `short_shift_duration` and `long_shift_duration`?

Reply:
Somewhat related: how do you use this weight? Is there a penalty for each time a `short_shift` is used?  Or for each created `short_shift`?

Reply:
This is used to weight the number of time periods a shift deviates from ideal shift duration (if a shift has either shorter or longer duration than ideal). So very short or long shifts are penalized harder than shifts that only have a small deviation. The weight is currently applied for both excess and deficit durations, but could (and perhaps should) be separated into two weights, differentiating between too short shifts and too long shifts.